includes/Modules/Search_Console.php

```diff
@@ -246,6 +246,20 @@ protected function create_data_request( Data_Request $data ) {
 		return new WP_Error( 'invalid_datapoint', __( 'Invalid datapoint.', 'google-site-kit' ) );
 	}
 
+	/**
+	 * Returns a punycode version of a unicode URL.
+	 *
+	 * @param string $url The URL to decode.
+	 */
+	protected function decode_unicode_url( $url ) {
+		$parts = wp_parse_url( $url );
+		if ( ! $parts ) {
+			return $url;
+		}
+		$decoded = \Requests_IDNAEncoder::encode( $parts['host'] );
+		return $parts['scheme'] . '://' . $decoded . $parts['path'];
```

Comment:
Maybe it would be "safer" to use `$host = wp_parse_url( $url, PHP_URL_HOST )`, then encode `$host` and use `str_replace( $host, $encoded_host, $url )`? Just wondering. There may be other parts, or there may not be a path, so I think only looking at the host is a safer bet.

Reply:
Sure, that makes sense. I was a little nervous about this approach given the various forms URLs take. I'll change this.